File: js/all-longest-strings.js
function allLongestString(inputArray){
    let maxLetterCount=0
    let letterCount=[]
    let result=[]
    for(i=0;i<inputArray.length;i++){
        letterCount[i]=inputArray[i].length
    }

    maxLetterCount=Math.max(...letterCount)

    result=inputArray.filter(x=>x.length==maxLetterCount)

    return result
}

let inputArray = ["aba", "aa", "ad", "vcd", "aba"]
console.log(allLongestString(inputArray))

File: go/all-longest-strings/all-longest-strings.go
package main

import(
	"fmt"
)

func allLongestStrings(inputArray []string)[]string{
	maxLetterCount := 0
	var res []string
	for _, val := range inputArray{
		if len(val) > maxLetterCount{
			maxLetterCount = len (val)
		}
	}

	for _, val := range inputArray{
		if len(val) == maxLetterCount{
			res = append(res, val)
		}
	}
	return res
}

func main() {
	inputArray := []string{"aba", "aa", "ad", "vcd", "aba"}
	fmt.Printf("The answer for the array %v is %v", inputArray, allLongestStrings(inputArray))
}